step1 웹 뷰 생성 (라자냐)

wonhee009 opened the pull request:
WKWebView를 스토리보드에 배치한 후 원하는 url를 웹뷰에 로드하도록 구현했습니다.
전체적으로 이름 짓기에 좀 신경을 썼습니다.
1. top constraint를 superView의 top에 걸지 safeArea의 top에 걸지 고민했는데 현재 프로젝트에는 navigationBar에 대한 언급이 없어 safeArea의 top에 constraint를 걸어 x 이후 모델의 노치 부분은 침범하지 않도록 구현했습니다.
2. 기능을 분리하여 func로 구현했습니다.
3. 에러 alert func는 현재 에러가 1개이므로 func내에서 처리하지만 step을 진행하며 에러가 늘어날 경우 message를 받아서 func는 alert만 생성하도록 바뀔 예정입니다.
4. 웹뷰를 로드하는 과정 중에서 로드 시간으로 인해 잠시 흰색 화면이 뜨는데 이는 다음 step에서 indicator를 추가할 생각입니다.

wonhee009 committed:
1. add setWebView func

O-O-wl commented on WebBrowser/WebBrowser/ViewController.swift:
```diff
@@ -5,14 +5,35 @@
 
 class ViewController: UIViewController {
+    
+    @IBOutlet weak var webView: WKWebView!
```
실제 Device에서 혹시 App을 실행해보셨나요? 
안 해보셨으면 한번 해보시면 좋을 것 같습니다🙇‍♂️

wonhee009 replied:
실제 Device와 Simulator에서 실행한 캡쳐화면인데 혹시 실행이 안되나요?

<img width="455" alt="스크린샷 2020-11-10 22 42 40" src="https://user-images.githubusercontent.com/30456206/98681449-13489800-23a6-11eb-9522-dc3d5d580085.png">

![IMG_3386-2](https://user-images.githubusercontent.com/30456206/98681659-5571d980-23a6-11eb-900b-b72e490db518.png)

O-O-wl replied:
Framework에 WebKit 추가하지 않고 인터페이스 빌더에서 WKWebView가 이니셜라이즈 될 때, 해당 클래스를 못 찾아오는 이슈가 있어서 코멘트드렸는데, 정상동적하고 있군요  😅😅

O-O-wl commented on WebBrowser/WebBrowser/ViewController.swift:
```diff
@@ -5,14 +5,35 @@
+    
+    func setWebView() {
+        guard let url = URL(string: myAddress) else {
+            return makeErrorAlert()
```
멋지네요!
guard let으로 안전하게 언래핑하는 것도 중요하지만 이렇게 실패에 대한 핸들링을 추가하는 것도 중요합니다 👍

O-O-wl commented on WebBrowser/WebBrowser/ViewController.swift:
```diff
@@ -5,14 +5,35 @@
+    
+    @IBOutlet weak var webView: WKWebView!
+    
+    private let myAddress = "https://www.indiepost.co.kr"
```
이 프로퍼티는
private으로 선언해두셨군요
혹시 다른 프로퍼티나 메소드의 access control은 private이 아닌 이유가 있을까요? 🤔

O-O-wl commented on WebBrowser/WebBrowser/ViewController.swift:
```diff
@@ -5,14 +5,35 @@
 
     override func viewDidLoad() {
         super.viewDidLoad()
         // Do any additional setup after loading the view.
```
추가적인 setup을 하라는 기본으로 생성되는 주석이네요 이부분은 이제 삭제해도 좋을 것 같습니다. 🗑

O-O-wl commented on WebBrowser/WebBrowser/ViewController.swift:
```diff
@@ -5,14 +5,35 @@
+    
+    @IBOutlet weak var webView: WKWebView!
+    
+    private let myAddress = "https://www.indiepost.co.kr"
```
주소를 프로퍼티로 선언하셨네요

이 문자열은 해당 ViewController가 사라질때 까지 유지되야 하는 값일까요?🤔

wonhee009 replied:
지금 step에서는 유지되어야 하는 값이 아니라고 생각되는데 이후 step에서는 주소를 프로퍼티로 가지고 있어야한다고 생각했습니다.😀
이후에 사용자에게 입력받은 string을 통해 url을 생성하고 그런 과정에서 error를 핸들링할때 다시 url 생성 요청시 이전에 입력받은 string을 통해 다시 생성해야할 수 있다고 생각했어요.
근데 이 부분은 이후에 step을 진행하면서 바뀔 수 있을거 같아요.

O-O-wl replied:
아하 추후 있을 변경에 대해 고민하신 흔적이군요

추후 확장을 하기 쉽게 구현하는 것도 중요하지만
현재 스펙에서 최선의 코드를 작성하려고 노력해보시면 좋을거 같아요 고생하셨습니다 😎

O-O-wl reviewed (commented):
🎉멋집니다.
스펙으로 정의되지 않은 URL이 유효하지 않을 때에 대한 핸들링까지 구현되어 있는게 인상적이네요.  작은 스펙이지만 커밋을 잘게 나누어보는 것도 한번 생각해보세요💬 
몇가지 코멘트 남겼습니다

O-O-wl commented on WebBrowser/WebBrowser/ViewController.swift:
```diff
@@ -5,14 +5,35 @@
+        webView.load(request)
+    }
+    
+    func makeErrorAlert() {
```
~make라는 네이밍은 객체를 생성한다는 의도를 표현하는 것 같아요.
show와 같은 네이밍은 어떨까요?~

<img width="800" alt="스크린샷 2020-11-10 오후 11 12 16" src="https://user-images.githubusercontent.com/39197978/98684987-2fe6cf00-23aa-11eb-8bb0-e2b03d04eee0.png">

이러한 관례가 있어서 오해의 여지가 있다고 생각해서 **주관적으로** 해당 코멘트 달았습니다 🙇‍♂️
제 리뷰도 하나의 의견이니 그냥 참고만 해주세요🙇‍♂️

wonhee009 committed:
delete setup 주석

wonhee009 committed:
2. delete setup 주석

O-O-wl reviewed (commented):
추가로 하나 코멘트 추가했습니답

yagom commented:
리뷰어의 질문에 대해 의논하고 싶은 부분은 코멘트에서 의논하면 되고, 리뷰 내용을 반영하여 다시 리뷰를 받고 싶은 경우에는 리뷰어를 태그하여 코멘트 하면 됩니다. 
@wonhee009 @O-O-wl

wonhee009 committed:
edit func name

make- -> show-로 변경
make는 객체를 반환해주는 prefix로 사용하는게 더 맞는거 같아요.
제가 의도한 func name은 객체를 return하는게 아니라 alert을 보여주는거뿐이라서 show가 더 적절하다 생각합니다.

wonhee009 committed:
edit func name

make- -> show-로 변경
make는 객체를 반환해주는 prefix로 사용하는게 더 맞는거 같아요.
제가 의도한 func name은 객체를 return하는게 아니라 alert을 보여주는거뿐이라서 show가 더 적절하다 생각합니다.

wonhee009 committed:
Merge remote-tracking branch 'origin/step1-lasagna' into step1-lasagna

O-O-wl commented on WebBrowser/WebBrowser/ViewController.swift:
```diff
@@ -20,14 +20,14 @@ class ViewController: UIViewController {
     func setWebView() {
         guard let url = URL(string: myAddress) else {
-            return makeErrorAlert()
+            return showErrorAlert()
```
```swift
guard let url = URL(string: myAddress) else {
      showErrorAlert()
      return
}
```
이런식의 표현만 써왔었는데
이런식의 표현이 가능한지 이번에 알게 되었네요 😮 
배워감미당

O-O-wl reviewed (commented):
고생하셨어요~ 머지 하겠습니다!
다음 Step 진행 하시면서 더 재밌는 시도들 많이 해보시면 좋을 것 같아요 🙇‍♂️

make -> show 네이밍은 주관적인 하나의 의견으로 봐주세요
라자냐 고생하셨습니다!
